src/bin/ion/commands/beta/head.rs

```diff
@@ -0,0 +1,54 @@
+use std::io::{stdout, Write};
+use clap::{Arg, ArgMatches, Command};
+use ion_rs::*;
+use std::fs::File;
+use anyhow::{Context, Result};
+use crate::commands::dump;
+
+pub fn app() -> Command {
+    Command::new("head")
+        .about("Prints the specified number of top-level values in the input stream.")
+        .arg(
+            Arg::new("values")
+                .long("values")
+                .short('n')
+                .allow_negative_numbers(false)
+                .help("Specifies the number of output top-level values.")
+        )
+        .arg(
+            Arg::new("format")
+                .long("format")
+                .short('f')
+                .default_value("lines")
+                .value_parser(["binary", "text", "pretty", "lines"])
+                .help("Output format"),
+        )
+        .arg(
+            Arg::new("input")
+                .required(true)
+                .help("Input file")
+        )
+}
+
+pub fn run(_command_name: &str, matches: &ArgMatches) -> Result<()> {
+    let input_file = matches
+        .get_one::<String>("input")
+        .unwrap();
+    let values: i32 = matches
+        .get_one::<String>("values")
+        .unwrap()
+        .parse()
+        .unwrap();
+    let format = matches
+        .get_one::<String>("format").
+        unwrap();
+    let file = File::open(input_file)
+        .with_context(|| format!("Could not open file '{}'", input_file))?;
+    if values == 0 {
+        return Ok(());
+    }
+    let mut reader = ReaderBuilder::new().build(file)?;
+    let mut output: Box<dyn Write> = Box::new(stdout().lock());
+    dump::write_all_in_format(&mut reader, &mut output, format, values).expect("Could not process the input stream.");
+    Ok(())
```

Comment:
Here's a crazy idea... why not just delegate to `dump::run`? You can even handle the `values` argument there, the other arguments are the same. Seems like you should be able to pass `matches` straight in from here to there. 

Then you get all the stdin and file handling logic from `dump`, for free.